### PR TITLE
Update beta mentions in Provider docs

### PR DESCRIPTION
**Pull request by upodroid**
@melinath

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by melinath** [comment]
Reassigning to Riley because this deals with details about how we message the two versions of the provider.

File: mmv1/third_party/terraform/website/docs/guides/provider_versions.html.markdown [hunk 1]
```diff
@@ -8,7 +8,7 @@ description: |-
 
 # Google Provider Versions
 
-Starting with the `1.19.0` release, there are two versions of the Google
+Starting with the `1.19.0` release, there are two versions of the Google Cloud
```
Comment (rileykarson) [inline mmv1/third_party/terraform/website/docs/guides/provider_versions.html.markdown:11]: "Google provider" or "Google Cloud Platform provider" is fine, but I'd avoid "Google Cloud provider" as we only support GCP. Google Workspace is part of Google Cloud but not Google Cloud Platform, for example, and is supported through a separate provider.

File: mmv1/third_party/terraform/website/docs/guides/provider_versions.html.markdown [hunk 2]
```diff
@@ -19,18 +19,13 @@ This documentation (https://registry.terraform.io/providers/hashicorp/google/lat
...
-supports GCP products and features that are in beta, while `google` does not.
-Fields and resources that are only present in `google-beta` will be marked as
-such in the shared provider documentation.
+You may see beta features referenced as Preview since Google simplified the [product launch stages](https://cloud.google.com/blog/products/gcp/google-cloud-gets-simplified-product-launch-stages) in late 2020.
 
-`1.X` series releases of the `google` provider supported beta features; from
-`2.0.0` onwards, beta features are only supported in `google-beta`.
+The `google-beta` provider is distinct from the `google` provider in that it
+supports GCP products and features that are in Preview, while `google` does not.
+Fields and resources that are only present in `google-beta` are marked clearly in the provider documentation.
```
Comment (rileykarson) [inline mmv1/third_party/terraform/website/docs/guides/provider_versions.html.markdown:26]:
```suggestion
Fields and resources that are only present in `google-beta` are clearly marked in the provider documentation.
```

File: mmv1/third_party/terraform/website/docs/guides/provider_versions.html.markdown (hunk 1 above)
Comment (upodroid) [reply mmv1/third_party/terraform/website/docs/guides/provider_versions.html.markdown:11]: GCP has been disappearing in favour of Google Cloud brand recently, but I added Platform anyway.

**Comment by upodroid** [comment]
bump

File: mmv1/third_party/terraform/website/docs/guides/provider_versions.html.markdown [hunk 3]
```diff
@@ -39,7 +34,7 @@ features.
 -> In short, using `google-beta` over `google` is similar to using `gcloud beta`
 over `gcloud`. Features that are exclusively available in `google-beta` are GCP
 features that are not yet GA, and they will be made available in `google` after
-their GA launch.
+they are in General Availability.
```
Comment (rileykarson) [inline mmv1/third_party/terraform/website/docs/guides/provider_versions.html.markdown:37]:
```suggestion
they go to GA.
```

nit: We've introduced GA earlier in the page and used it numerous times, it's more consistent to use here imo.

Comment (upodroid) [reply mmv1/third_party/terraform/website/docs/guides/provider_versions.html.markdown]: Fixed

**Review by rileykarson — APPROVED** [review]

**Pull request merged**